apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts

```diff
@@ -0,0 +1,209 @@
+import { MapEdit, MapTypesList } from './MapConstructor'
+import { notValidBuildingMapJson, buildingMapJson } from '@condo/domains/property/constants/property.example'
+import { cloneDeep } from 'lodash'
+
+const testSection = {
+    id: '',
+    minFloor: -5,
+    maxFloor: 10,
+    unitsOnFloor: 10,
+}
+
+const sectionName = () => {
+    return Math.random().toString()
+}
+const createBuildingMap = (sections: number) => {
+    const PropertyMap = new MapEdit(null, () => null )
+    for (let i = 0; i < sections; i++){
+        PropertyMap.addSection({ ...testSection, name: sectionName(), type: MapTypesList.Section })
+    }
+    return PropertyMap
+}
+
+
+describe('Map constructor', () => {
```

Comment:
could you please provide structure tests like in BBuilderData? It's hard to understand is it backward compatible or not. Without the state tests it's hard to understand that logic too

Reply:
1. Moved old JSONS from BBuilderData to new tests
2. Added new logic to auto repair mechanics (as old JSON misses some key fields like name, dv, type ....) 
3. Added tests to check result structure to match JSON structure
4. Old tests have some checks on old matrix mechanics which I didn't implement so I skip these tests